modules/extensions-support/src/main/java/org/mule/module/extension/internal/manager/DefaultExtensionManager.java

```diff
@@ -0,0 +1,425 @@
+
+        if (implicitConfiguration == null)
+        {
+            throw new IllegalStateException(String.format("Could not find a config for extension %s and none can be created automatically. Please define one", extension.getName()));
+        }
+
+        ConfigurationObjectBuilder configurationObjectBuilder = new ConfigurationObjectBuilder(implicitConfiguration, buildImplicitConfigurationResolverSet(implicitConfiguration));
+
+        Object configurationInstance;
+        try
+        {
+            configurationInstance = configurationObjectBuilder.build(asOperationContextAdapter(operationContext).getEvent());
+        }
+        catch (MuleException e)
+        {
+            throw new MuleRuntimeException(e);
+        }
+
+        final String instanceName = objectNameHelper.getUniqueName(String.format("%s-%s", extension.getName(), implicitConfiguration.getName()));
+        registerConfigurationInstanceProvider(instanceName,
+                                              new StaticConfigurationInstanceProvider<>(instanceName, implicitConfiguration, configurationInstance));
+
+        return configurationInstance;
+    }
+
+    private ResolverSet buildImplicitConfigurationResolverSet(Configuration configuration)
+    {
+        ResolverSet resolverSet = new ResolverSet();
+        for (Parameter parameter : configuration.getParameters())
+        {
+            Object defaultValue = parameter.getDefaultValue();
+            if (defaultValue != null)
+            {
+                ValueResolver<Object> valueResolver;
+                if (defaultValue instanceof String && muleContext.getExpressionManager().isExpression((String) defaultValue))
+                {
+                    valueResolver = new EvaluateAndTransformValueResolver<>((String) defaultValue, parameter.getType());
+                }
+                else
+                {
+                    valueResolver = new StaticValueResolver<>(defaultValue);
+                }
+
+                resolverSet.add(parameter, valueResolver);
+            }
+        }
+
+        return resolverSet;
+    }
+
+    private Configuration getImplicitConfiguration(Extension extension)
+    {
+        for (Configuration configuration : extension.getConfigurations())
+        {
+            if (canBeUsedImplicitly(configuration))
+            {
+                return configuration;
+            }
+        }
+
+        return null;
+    }
+
+    private boolean canBeUsedImplicitly(Configuration configuration)
+    {
+        for (Parameter parameter : configuration.getParameters())
+        {
+            if (parameter.isRequired() && parameter.getDefaultValue() == null)
+            {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private boolean maybeUpdateExtension(Extension extension, String extensionName)
+    {
+        Extension actual = register.getExtension(extensionName);
+        MuleVersion newVersion;
```

Comment:
same answer as before, what you say is a possible behaviour, but not the one defined in the use case that drove this particular piece. 

The case of a patch was not accounted for, but I believe that patchers require a restart to take effect so that would fall under standard case